File: src/main/java/ru/pashkovske/buratino/tinkoff/service/assignment/AssignmentDao.java
package ru.pashkovske.buratino.tinkoff.service.assignment;

import lombok.NonNull;
import ru.pashkovske.buratino.tinkoff.service.instrument.model.InstrumentId;
import ru.tinkoff.piapi.contract.v1.OrderDirection;

import java.util.List;
import java.util.UUID;

public interface AssignmentDao {
    void post(@NonNull Assignment assignment);
    void update(@NonNull UUID id, @NonNull Assignment assignment);
    Assignment get(@NonNull UUID id);
    List<Assignment> findByInstrument(@NonNull InstrumentId instrumentId);
    List<Assignment> findByInstrumentAndDirection(@NonNull InstrumentId instrumentId, @NonNull OrderDirection direction);
    void delete(@NonNull UUID id);
}

File: src/main/kotlin/ru/pashkovske/buratino/assignment/base/dao/AssignmentDao.kt
package ru.pashkovske.buratino.assignment.base.dao

import ru.pashkovske.buratino.assignment.base.model.Assignment
import ru.pashkovske.buratino.assignment.base.model.AssignmentState
import java.util.UUID

interface AssignmentDao<A : Assignment> {
    fun getAll(): List<A>
    fun getByStatus(status: AssignmentState): List<A>
    fun get(id: UUID): A
    fun create(assignment: A)
    fun update(assignment: A)
    fun deleteAll()
}